modules/programs/zsh.nix

```
@@ -495,6 +505,7 @@ in
         # See https://github.com/nix-community/home-manager/issues/177.
         HISTSIZE="${toString cfg.history.size}"
         SAVEHIST="${toString cfg.history.save}"
+        ${if cfg.history.ignorePatterns != [] then "HISTORY_IGNORE=${lib.escapeShellArg "(${lib.concatStringsSep "|" cfg.history.ignorePatterns})"}" else ""}
```

Maybe using `optionalString` would be cleaner?

```suggestion
        ${optionalString (cfg.history.ignorePatterns != []) "HISTORY_IGNORE=${lib.escapeShellArg "(${lib.concatStringsSep "|" cfg.history.ignorePatterns})"}"}
```

---

Fixed.